gridstatus/isone_api/isone_api.py

```diff
@@ -382,3 +341,201 @@ def get_dayahead_hourly_demand(
         df = df.drop_duplicates(subset=["BeginDate", "Location"], keep="first")
 
         return self._handle_demand(df, interval_minutes=60)
+
+    def _handle_demand(
+        self,
+        df: pd.DataFrame,
+        interval_minutes: int = 60,
+    ) -> pd.DataFrame:
+        """
+        Process demand DataFrame: convert types, rename columns, and add Interval End.
+
+        Args:
+            df (pd.DataFrame): Input DataFrame with demand data.
+            interval_minutes (int): Duration of each interval in minutes. Default is 60.
+
+        Returns:
+            pd.DataFrame: Processed DataFrame.
+        """
+        try:
+            # Try the standard pandas datetime conversion first
+            df["Interval Start"] = pd.to_datetime(df["BeginDate"])
+            df["Interval Start"] = df["Interval Start"].dt.tz_convert(
+                self.default_timezone,
+            )
+        except AttributeError:
+            # NOTE(kladar) consistently the above logic fails for DST conversion days.
+            # This catches those and fixes it.
+            # Data coming out looks good, no missed intervals and no duplicates.
+            log.warning("Standard datetime conversion failed. Using custom parsing.")
+            df["Interval Start"] = df["BeginDate"].apply(
+                self.parse_problematic_datetime,
+            )
+
+        df = df.sort_values(["Interval Start", "Location"])
+        df["Interval End"] = df["Interval Start"] + pd.Timedelta(
+            minutes=interval_minutes,
+        )
+
+        df["Load"] = pd.to_numeric(df["Load"], errors="coerce")
+        df["Location Id"] = pd.to_numeric(df["LocId"], errors="coerce")
+
+        log.info(
+            f"Processed demand data: {len(df)} entries from {df['Interval Start'].min()} to {df['Interval Start'].max()}",
+        )
+        return df[["Interval Start", "Interval End", "Location", "Location Id", "Load"]]
+
+    @support_date_range("DAY_START")
+    def get_hourly_load_forecast(
+        self,
+        date: str | pd.Timestamp = "latest",
+        end: str | pd.Timestamp | None = None,
+        horizons: Literal["latest", "all"] = "all",
```

Comment:
Sorry, should have mentioned this earlier, but we call different versions of a forecast "vintages" so maybe we should use that here. I've only heard "horizon" used in the context of how far out a forecast goes ([example](https://help.anaplan.com/forecast-horizon-calculations--b7c8bc9e-ff41-4705-92fa-c94ec1cf7bdf)).

@kmax12 Do you have an opinion?

Reply:
Yep makes sense, I think of horizon and vintage as two sides of the same coin so happy to go with either where it makes sense.

`vintage` ~= `publish_time` aka, when was a forecast made? (like wine vintages. If a vintage is "2021", it was made in 2021)
`horizon` ~= `interval - publish_time` aka, how far out is a "forecast" (or collated series of forecasted intervals) 

In this case the `latest` actually does return a `horizon`, rather than a `vintage`. Typically, for relevant forecast performance statistics, we want the same horizon, basically "how good are we at predicting 30 minutes out? 1 day out?" It's collated across publish times or combining multiple vintages.

Comparing across vintages typically just shows how your forecast production system is improving over time. Which can be useful, but is technically not what is going on here, from my perspective.  

I talked about this in my case study briefly. 
![CleanShot 2024-10-21 at 08 35 53@2x](https://github.com/user-attachments/assets/5f03adac-1f88-43c6-b9a3-bdc836ffb261)

Reply:
Neither here nor there, but I think we should add a horizon column to all forecast data, because it is an incredibly useful index and a bit of an expensive operation to create it sometimes.